drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -290,6 +290,28 @@ def normalize_single_custom_field_value(cded, value)
       else
         raise "unexpected value \"#{value}\""
       end
+    when 'file'
```

Comment:
Overall, is this the right place for this logic to go? I was wondering if there's something cleverer I could do with the [FileProcessor](https://github.com/greenriver/hmis-warehouse/blob/release-142/drivers/hmis/app/models/hmis/hud/processors/file_processor.rb), but I wasn't sure how to adapt that processor, which is geared towards processing individual attributes onto a File record, to make it compatible with this new usage, which is geared towards processing a Custom Data Element that _points_ at a File record.

Reply:
This is inside of the normalization function, but this seems to really go above and beyond normalizing to actually process the blob id into a file object.

Maybe we can call this separately after normalization occurs, something like:
```ruby
    # Normalize the input value
    value = normalize_custom_field_value(cded, value)
    # Transform file blob id into (new or existing?) File. We will save the reference to this File on the CDE
    value = process_blob_id_to_file(cded, value) if cded.field_type == :file
```